dwave_networkx/algorithms/partition.py

```diff
@@ -0,0 +1,157 @@
+# Copyright 2021 D-Wave Systems Inc.
+#
+#    Licensed under the Apache License, Version 2.0 (the "License");
+#    you may not use this file except in compliance with the License.
+#    You may obtain a copy of the License at
+#
+#        http://www.apache.org/licenses/LICENSE-2.0
+#
+#    Unless required by applicable law or agreed to in writing, software
+#    distributed under the License is distributed on an "AS IS" BASIS,
+#    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+#    See the License for the specific language governing permissions and
+#    limitations under the License.
+
+from dimod import ConstrainedQuadraticModel
+from dimod import quicksum 
+from dimod import Binary
+import numpy as np
+from math import isclose
+from itertools import product

```

Comment:
[Order of imports](https://www.python.org/dev/peps/pep-0008/#imports). Also, IMO it keeps the namespace a lot cleaner to just do
```python
import itertools
import math

import dimod
```

and then do `dimod.Binary(..)`, `math.isclose(..)` etc.